psycopg_pool/psycopg_pool/null_pool_async.py

```diff
@@ -29,7 +29,7 @@ def __init__(
         self: AsyncNullConnectionPool[AsyncConnection[TupleRow]],
         conninfo: str = "",
         *,
-        open: bool = ...,
+        open: bool | None = ...,
```

Comment:
I'm not reviewing this PR but just a couple of random drive-by typing questions:
1. Is the type annotation for `self` needed? (I didn't think it generally was, since the type checker knows what type the object is?)
2. What does this ellipsis assignment syntax mean? I've not come across this before.

Reply:
Annotation for `self` is not needed in general when used in a normal method.

But here, this is an `@overload` of the `__init__()` method in which the "constructed" object, `self`, will vary depending on other arguments to the constructor: the type of `self` depends on the value of `connection_class` argument. So by annotating `self` along with `connection_class` in each overload definition, we declare this type mapping.

The ellipsis for parameter (`= ...`) in `@overload` definitions is used when the parameter has a default value in order to avoid repeating it in each definitions whereas it's already declared in the non-overloaded definition (the last one): https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values

Reply:
Thanks for the clarifications @dlax!
(I should have looked at more context than what was shown in the diff, to notice that this was an `@overload` definition 🤦 )

Reply:
Thank you for your answer, @dlax, you beat me to it.

In effect, as `__init__` doesn't return anything, the annotation on `self` is used in place as an annotation on the return value.